### PR TITLE
Introduce disabling for Touchable* elements

### PR DESCRIPTION
**Pull request by Kureev**
Introduce a `disabled` property for Touchable* components.

Fix https://github.com/facebook/react-native/issues/2103

### CODE REVIEW

**Comment by satya164** [comment]
cc @mkonicek

**Comment by Kureev** [comment]
@facebook-github-bot liar!

File: Libraries/Components/Touchable/TouchableWithoutFeedback.js [hunk 1]
```diff
@@ -41,6 +41,10 @@ var TouchableWithoutFeedback = React.createClass({
       React.PropTypes.arrayOf(React.PropTypes.oneOf(View.AccessibilityTraits)),
     ]),
     /**
+     * Disable all component interactions if value is true
```
Comment (mkonicek) [inline Libraries/Components/Touchable/TouchableWithoutFeedback.js]: nit: if the value is true / If true, disable all interactions for this component.

**Comment by mkonicek** [comment]
Does this work also for `TouchableHighlight`, `TouchableBounce`, `TouchableNativeFeedback` etc.? 

Do they still animate if you set disabled to true? Do you want them to animate? Should `TouchableNativeFeedback` (Android-only) animate when disabled?

You could try this in the [`TouchableExample`](https://github.com/facebook/react-native/blob/master/Examples/UIExplorer/TouchableExample.js) and ideally add examples there if this approach works well and disables the animations.

**Comment by Kureev** [comment]
Will do!

**Comment by mkonicek** [comment]
:+1: thank you.

**Comment by Kureev** [comment]
@mkonicek I've added a disabled example to UIExplorer:
image
It works with all Touchables (all of them inherits from TouchableWithoutFeedback which use Touchable mixin [I've patched by disabled flag](https://github.com/facebook/react-native/pull/5931/files#diff-580446e098bcae0cd35585b8cf689f2aR340))

File: Examples/UIExplorer/TouchableExample.js [hunk 2]
```diff
@@ -292,6 +299,45 @@ var TouchableHitSlop = React.createClass({
...
+          <Text style={styles.button}>Enabled TouchableOpacity</Text>
+        </TouchableOpacity>
+
+        <TouchableHighlight
+          activeOpacity={1}
+          disabled={true}
+          animationVelocity={0}
+          underlayColor="rgb(210, 230, 255)"
+          style={[styles.row, styles.block]}
+          onPress={() => console.log('custom THW text - hightlight')}>
```
Comment (mkonicek) [inline Examples/UIExplorer/TouchableExample.js]: nit: highlight

File: Examples/UIExplorer/TouchableExample.js [hunk 3]
```diff
@@ -292,6 +299,45 @@ var TouchableHitSlop = React.createClass({
...
+            Disabled TouchableHighlight
+          </Text>
+        </TouchableHighlight>
+
+        <TouchableHighlight
+          activeOpacity={1}
+          animationVelocity={0}
+          underlayColor="rgb(210, 230, 255)"
+          style={[styles.row, styles.block]}
+          onPress={() => console.log('custom THW text - hightlight')}>
```
Comment (mkonicek) [inline Examples/UIExplorer/TouchableExample.js]: nit: highlight

**Comment by mkonicek** [comment]
OK, let's shipit.

The test plan doesn't show whether this disables animations for `TouchableNativeFeedback` but we can fix that later if it still animates with this PR.

@facebook-github-bot shipit

**Comment by Kureev** [comment]
I'll make a 2 follow-up PRs: 
- Hightlight -> Highlight
- Add additional example with `TouchableNativeFeedback`

Thanks for merging this one in, @mkonicek

**Comment by mkonicek** [comment]
:100: Awesome, thanks @Kureev!

**Comment by Kureev** [comment]
haha, you're welcome, @mkonicek :wink:

**Pull request closed without merging**